Overhaul language definitions in preparation for GF_Glyphset langdef-based overhaul

yanone opened the pull request.

moyogo reviewed (commented):
This is getting tricky. Some of these are not actually required and some of these are only required in some orthographies of a single language that has multiple orthographies.

Ligature ij is not required to support Dutch but may be nice to have for typesetting. Fonts don't have to have it to support Dutch. I would hard code this in glyphssets when handling Dutch, not in the language data base exemplar set. Maybe we need additional data to represent this in gflanguages.

Sharp s is used in some Low German orthographies but not in others. So fonts may support some orthographies of Low German even if they don't have it. For this we need to allow multiple orthographies in gflanguages.

I don't think these are blocking this PR but they'll need to be addressed at some point soon because they misrepresent language requirements.

yanone commented:
@moyogo I removed the ij and IJ from Dutch.

I added the lowercase ß to Low German only because it already had an uppercase version defined previously, which on its own is certainly nonsensical.

RosaWagner commented:
As @moyogo suggested also in #84 `section sign` could also be removed

yanone commented:
Okay, I removed the section sign and all duplicate spaces

yanone commented:
Today I will put together a document that outlines in which order the various repos need to be merged and releases cut, so please wait with merging.

moyogo commented on Lib/gflanguages/data/languages/de_Latn.textproto:
```diff
@@ -33,11 +33,11 @@ region: "SK"
-  index: "A B C D E F G H I J K L M N O P Q R S T U V W X Y Z"
+  numerals: "- , . % + 0 1 2 3 4 5 6 7 8 9"
+  punctuation: "- – — , ; : ! ? . … \' ‘ ‚ \" “ „ « » ( ) [ ] { } @ * / & #"
+  index: "A B C D E F G H I J K L M N O P Q R S ẞ T U V W X Y Z"
```
ẞ shouldn’t be in the de_Latn index it isn’t sorted differently than SS.

RosaWagner replied:
Can we solve it by adding all the capitals in the base glyphs including capital eszett?

moyogo replied:
Yes, the capitals should be added to the base characters. The index is for indexing/sorting, unfortunately there’s no explanation on what it’s for here. I don’t think GF uses the index anywhere, it was imported from the CLDR which uses it for sorting. We need a different PR that adds the uppercase characters, it was done previously for some language data in #53 but not for all.

moyogo commented on Lib/gflanguages/data/languages/ltg_Latn.textproto:
```diff
@@ -8,5 +8,5 @@ region: "LV"
   base: "A Ā B C Č D E Ē F G Ģ H I Ī J K Ķ L Ļ M N Ņ O Ō P R S Š T U Ū V Y Z Ž a ā b c č d e ē f g ģ h i ī j k ķ l ļ m n ņ o ō p r s š t u ū v y z ž"
   auxiliary: "Q W X q w x"
-  marks: "◌̄ ◌̌ ◌̧"
+  marks: "◌̄ ◌̌ ◌̧ ◌ ̒"
```
Adding 0x0312 COMBINING TURNED COMMA ABOVE to Latvian (lv), Latgalian (ltg), Prussian (prg) is incorrect. While the glyph may be used as a component of ģ, the character is not used in those languages.

moyogo reviewed (commented):
Besides these comments, I think this can be merged.
